File: Data Structures & Algorithms/reconstruct-flight-path/submission-0.py
class Solution:
    def findItinerary(self, tickets: List[List[str]]) -> List[str]:
        adj = collections.defaultdict(list)
        tickets.sort()
        for u, v in tickets:
            adj[u].append(v)

        res = ["JFK"]
        def dfs(src):
            if len(res) == len(tickets) + 1:
                return True
            elif src not in adj:
                return False
            
            tmp = list(adj[src])
            for i, v in enumerate(tmp):
                adj[src].pop(i)
                res.append(v)

                if dfs(v): return True

                res.pop()
                adj[src].insert(i, v)
            return False

        dfs("JFK")
        return res

File: Data Structures & Algorithms/non-overlapping-intervals/submission-0.cpp
class Solution {
public:
    int eraseOverlapIntervals(vector<vector<int>>& intervals) {
        std::sort(intervals.begin(), intervals.end(), [&](vector<int>& a, vector<int>& b) {
            if (a[1] == b[1]) return a[0] > b[0];
            return a[1] < b[1];
        });

        int last_end{intervals[0][1]}, res{};
        for (int i = 1; i < intervals.size(); ++i) {
            if (last_end > intervals[i][0]) {
                ++res;
            }
            else {
                last_end = intervals[i][1];
            }
        }

        return res;
    }
};
